Documentation/bpf.rst

```diff
@@ -1865,7 +1865,7 @@ describe some of the differences for the BPF model:
   Another possibility is to use tail calls by calling into the same program
   again and using a ``BPF_MAP_TYPE_PERCPU_ARRAY`` map for having a local
   scratch space. While being dynamic, this form of looping however is limited
-  to a maximum of 32 iterations.
+  to a maximum of 34 iterations.
```

Comment:
For reviewers: AFAIU, 33 tail calls would allow for 34 loop iterations with this approach.

Reply:
Why not making it explicit?
```suggestion
  to a maximum of 34 iterations (the initial program, plus 33 iterations from the tail calls).
```

Reply:
The initial program *is* the iteration body, no? That's what I'm unsure about.

Reply:
I meant _initial program_ in the sense, the one which is originally called (from the hook and not from the tail call). Not sure how to designates it. My reading is that it can be the iteration body indeed (depending of course on how you implement it).

[I'm sure you noticed, and probably out of the scope of this PR, the whole list item is about not having loops other than via unrolling, which is outdated - but probably requires a larger update effort]